### PR TITLE
Add click events

### LINKED ISSUE #18: Add mouse click event
Add mouse click event on the map. The event could be used to retrieve the map coordinates.

### PR DESCRIPTION
**Pull request by ngonzalezpazFC**
Close #18

### CODE REVIEW

File: src/main/java/com/flowingcode/vaadin/addons/googlemaps/GoogleMap.java [hunk 1]
```diff
@@ -236,4 +245,63 @@ public int getMinZoom() {
...
+		public double getLongitude() {
+			return this.lon;
+		}
+		public GoogleMapRightClickEvent(GoogleMap source, boolean fromClient,
+				@EventData(value = "event.detail.latLng") JsonValue latLng)
+				throws JsonMappingException, JsonProcessingException {
+			super(source);
+			@SuppressWarnings("unchecked")
+			HashMap<String, Double> result = new ObjectMapper()
+					.readValue(latLng.toJson(), HashMap.class);
```
Comment (javier-godoy) [inline src/main/java/com/flowingcode/vaadin/addons/googlemaps/GoogleMap.java:294]: I understand that `latLng` is a JSON object. The following should work:
```
elemental.json.JsonValue latLng;
lat = ((JsonObject)latLng).getNumber("lat");
lng = ((JsonObject)latLng).getNumber("lng");
```

**Review by javier-godoy — CHANGES_REQUESTED** [review]

**Review by javier-godoy — APPROVED** [review]

**Comment by javier-godoy** [comment]
@ngonzalezpazFC Since GoogleMapClickEvent and GoogleMapRightClickEvent have both the same properties, does it make sense to introduce an abstract class `GoogleMapClickEvent` and have `GoogleMapLeftClickEvent` and `GoogleMapRightClickEvent` both extend from it?

**Comment by ngonzalezpazFC** [comment]
> 
> 
> @ngonzalezpazFC Since GoogleMapClickEvent and GoogleMapRightClickEvent have both the same properties, does it make sense to introduce an abstract class `GoogleMapClickEvent` and have `GoogleMapLeftClickEvent` and `GoogleMapRightClickEvent` both extend from it?

Yes, I think is a good idea to avoid code duplication. Should the abstract class be in a separated file?

**Comment by javier-godoy** [comment]
> > @ngonzalezpazFC Since GoogleMapClickEvent and GoogleMapRightClickEvent have both the same properties, does it make sense to introduce an abstract class `GoogleMapClickEvent` and have `GoogleMapLeftClickEvent` and `GoogleMapRightClickEvent` both extend from it?
> 
> Yes, I think is a good idea to avoid code duplication. Should the abstract class be in a separated file?

I think it's fine in the same file.

**Review by javier-godoy — COMMENTED** [review]
Would you consider renaming it as follows?
 AbstractGoogleMapClickEvent -> GoogleMapClickEvent
 GoogleMapClickEvent -> GoogleMapLeftClickEvent 

Even better, instead of mapping the event with annotations, perhaps we can do:
```
public Registration addClickListener(
	DomListenerRegistration registration = this.getElement().addEventListener("google-map-rightclick", ev->{
	JsonObject latLng = ev.getEventData().get("event.detail.latLng");
	listener.click(new GoogleMapClickEvent(this,true,latLng));
	}).addEventData("event.detail.latLng");
    return registration::remove;
}
```
That way we'll avoid using different classes for the left and right clicks.

**Pull request merged**